File: cloud-storage-spring-java/src/main/java/com/github/sevtech/cloud/storage/spring/property/AwsS3Properties.java
package com.github.sevtech.cloud.storage.spring.property;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;

@Getter
@Setter
@ConfigurationProperties(prefix = "aws.s3")
public class AwsS3Properties {

	private String accessKey;
	private String secretKey;
	private String region;

	private Boolean localstackEnabled;
	private String localstackEndpoint;
	private String localstackRegion;
}

File: cloud-storage-spring-kotlin/src/main/kotlin/com/github/sevtech/cloud/storage/spring/property/aws/AwsS3Properties.kt
package com.github.sevtech.cloud.storage.spring.property.aws

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "aws.s3")
class AwsS3Properties {
    var accessKey: String? = null
    var secretKey: String? = null
    var region: String? = null
    
    var localstackEnabled: Boolean = false
    var localstackEndpoint: String? = null
    var localstackRegion: String? = null
}